### PR TITLE
Replaced rllab.envs.Env with gym.Env

### PR DESCRIPTION
**Pull request by jonashen**
This PR creates two utility files (i.e. gym_env_util and gym_space_util) to convert gym spaces for use in rllab algorithms.

### CODE REVIEW

File: contrib/ros/envs/sawyer_env.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
+"""
```
Comment (ryanjulian) [inline contrib/ros/envs/sawyer_env.py]: unlikely you created this file. this must have snuck in.

File: rllab/algos/ddpg.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
 from functools import partial
-import pickle
+import pickle as pickle
```
Comment (ryanjulian) [inline rllab/algos/ddpg.py]: `as pickle` is useless. please remove.

File: tests/test_spaces.py [hunk 3]
```diff
@@ -1,34 +1,38 @@
+import gym
 import numpy as np
 
+from rllab.envs.util import flatten, flatten_n
+from rllab.envs.util import sample, unflatten, unflatten_n
 from rllab.spaces import Box
 from rllab.spaces import Discrete
 from rllab.spaces import Product
```
[the remark this answers is not in the document]
Comment (zhanpenghe) [reply tests/test_spaces.py]: I agree. I think we have enough utils to deal with spaces.

[the remark this answers is not in the document]
Comment (ryanjulian) [reply tests/test_spaces.py]: The Spaces are a pretty fundamental part of how the NN part works, so we would have to be careful.

In any case, making changes to Spaces is out-of-scope for this change.

File: rllab/envs/dm_control_env.py [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
+import gym
 import numpy as np
 import pygame
 
 from rllab.core import Serializable
-from rllab.envs import Env
 from rllab.envs import Step
 from rllab.envs.dm_control_viewer import DmControlViewer
-from rllab.spaces import Box
 from rllab.spaces import Discrete
```
[the remark this answers is not in the document]
Comment (ryanjulian) [reply rllab/envs/dm_control_env.py]: done

File: tests/test_spaces.py [hunk 5]
```diff
@@ -1,34 +1,38 @@
+import gym
```
Comment (ryanjulian) [inline tests/test_spaces.py]: This file makes no sense. We don't need to test gym.Spaces, it's assumed to work. rllab.Spaces still exists and should be tested.

We might want to test rllab.envs.util, but that's a different test.

File: docs/user/gym_integration.rst [hunk 6]
```diff
@@ -1,93 +0,0 @@
-.. _gym_integration:
```
[the remark this answers is not in the document]
Comment (ryanjulian) [reply docs/user/gym_integration.rst]: ack

File: tests/test_spaces.py (hunk 5 above)
Comment (ryanjulian) [reply tests/test_spaces.py]: restored.

File: rllab/algos/ddpg.py (hunk 2 above)
Comment (ryanjulian) [reply rllab/algos/ddpg.py]: ack

File: rllab/envs/mujoco/hill/terrain.py [hunk 7]
```diff
@@ -109,8 +110,7 @@ def save_texture(x, y, hfield, fname, path=None):
     plt.contourf(x, y, -hfield, 100, cmap=TERRAIN_CMAP)
     xmin, xmax = x.min(), x.max()
     ymin, ymax = y.min(), y.max()
-    # for some reason plt.grid does not work here, so generate gridlines
-    # manually
+    # for some reason plt.grid does not work here, so generate gridlines manually
```
[the remark this answers is not in the document]
Comment (ryanjulian) [reply rllab/envs/mujoco/hill/terrain.py]: ack

File: scripts/sim_env.py [hunk 8]
```diff
@@ -5,8 +5,7 @@
 import numpy as np
 import pygame
 
-from rllab.envs import Env
-from rllab.misc.resolve import load_class
+from rllab.misc import load_class
```
[the remark this answers is not in the document]
Comment (ryanjulian) [reply scripts/sim_env.py]: ack

File: tests/envs/test_envs.py [hunk 9]
```diff
@@ -68,7 +49,6 @@
 envs.append(NoisyObservationEnv(CartpoleEnv()))
 envs.append(DelayedActionEnv(CartpoleEnv()))
 envs.append(NormalizedEnv(CartpoleEnv()))
-envs.append(GymEnv('CartPole-v0'))
```
[the remark this answers is not in the document]
Comment (ryanjulian) [reply tests/envs/test_envs.py]: i see why

**Comment by ryanjulian** [comment]
why is GymEnv still in the repo?

File: examples/cluster_gym_mujoco_demo.py [hunk 10]
```diff
@@ -24,13 +26,12 @@ def seed(self):
 def run_task(vv):
 
     env = TfEnv(
-        normalize(
-            GymEnv('HalfCheetah-v1', record_video=False, record_log=False)))
+        gym.make('HalfCheetah-v1', record_video=False, record_log=False))
```
Comment (ryanjulian) [inline examples/cluster_gym_mujoco_demo.py]: `gym.make()` does not accept kwargs

File: examples/trpo_gym_tf_cartpole.py [hunk 11]
```diff
@@ -11,12 +13,12 @@
 
 # Need to wrap in a tf environment and force_reset to true
 # see https://github.com/openai/rllab/issues/87#issuecomment-282519288
-env = TfEnv(normalize(GymEnv("CartPole-v0", force_reset=True)))
+env = TfEnv(gym.make("CartPole-v0", force_reset=True))
```
Comment (ryanjulian) [inline examples/trpo_gym_tf_cartpole.py]: `gym.make()` does not accept kwargs

**Pull request merged**